src/SlackThreads.jl

```diff
@@ -5,14 +5,17 @@ using StructTypes
 using FileIO
 using Mocking
 
-export SlackThread, slack_log_exception
+export AbstractSlackThread, SlackThread, slack_log_exception
+export DummyThread, SlackCallRecord
 
-mutable struct SlackThread
+abstract type AbstractSlackThread end
+
+mutable struct SlackThread <: AbstractSlackThread
     channel::Union{String,Nothing}
     ts::Union{String,Nothing} # In Slack terminology, `ts_thread`: the ID of another un-threaded message to reply to (https://api.slack.com/reference/messaging/payload)
 end
 
-function Base.copyto!(thread::SlackThread, obj)
+function Base.copyto!(thread::AbstractSlackThread, obj)
```

Comment:
Should this be implemented on a per-concrete type basis?

Reply:
true, done